### PR TITLE
Iz gawb 2645 optimize loading

### PR DESCRIPTION
**Pull request by zarsky-broad**
### Changes
- React is loaded from NPM
- Updated Webpack & plugins to current versions
- Updated version of Clojurescript
- Updated `dmohs/react`
- Better cljs compiler flags for dev & deploy
  - Should run faster when deployed, be clearer when debugging dev
- Js dependencies are broken into bundles & only loaded when needed
  - Style dependencies are placed into `<style>` tags in the DOM automatically when needed
- Renamed state colors from `:x-state` to `:state-x` for ease of use
- Created `ScriptLoader` using [react-load-script](https://github.com/blueberryapps/react-load-script)
- Broke `EntityDetails` out of `components` to avoid circular dependency
- Replaced `Infinity` and `-Infinity` with `js/Infinity` and `js/-Infinity` for new cljs
- **Timing diagrams, Markdown rendering, Codemirror, and IGV now load only when needed**

---

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Review by MattPutnam — APPROVED** [review]
👍 LGTM

File: project.clj [hunk 1]
```diff
@@ -38,7 +38,11 @@
                   ;; infinite recursion, which I was not able to figure
                   ;; out.
                   :optimizations :simple
```
Comment (davidangb) [inline project.clj:40]: did you try `:optimizations :advanced`? Update: bah, I tried it, and it causes an error:
```
Uncaught TypeError: ReactDOM.re is not a function
    at MB (compiled.js?t=1508518348002:734)
    at compiled.js?t=1508518348002:2445
    at compiled.js?t=1508518348002:2446
```

**Review by davidangb — COMMENTED** [review]
On first load, I see a warning in console:
```
Warning: It looks like you're using a minified copy of the development build of React. When deploying React apps to production, make sure to use the production build which skips development warnings and is faster. See https://fb.me/react-minification for more details.
```
Other than that, everything looks fine. Since this change is really pervasive, we just need to make sure it gets extra QA attention from all involved.

File: src/cljs/main/broadfcui/common/codemirror.cljs [hunk 2]
```diff
@@ -62,25 +68,34 @@
...
        (swap! locals assoc :code-mirror-component
-              (js-invoke CodeMirror-js "fromTextArea" (@refs "code-text")
+              (js-invoke (get-codemirror) "fromTextArea" (@refs "code-text")
                          #js{:mode mode :lineNumbers line-numbers? :readOnly read-only?
                              :viewportMargin js/Infinity}))))
    :component-will-receive-props
-   (fn [{:keys [props next-props locals]}]
-     (when (:read-only? props)
+   (fn [{:keys [props next-props state locals]}]
+     (when (and (:read-only? props) ())
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/common/codemirror.cljs]: Clean up

File: src/cljs/main/broadfcui/components/entity_details.cljs [hunk 3]
```diff
@@ -0,0 +1,84 @@
+(ns broadfcui.components.entity-details
+  (:require
+   [dmohs.react :as react]
+   [clojure.string :as string]
+   [broadfcui.common :as common]
+   [broadfcui.common.codemirror :refer [CodeMirror]]
+   [broadfcui.common.style :as style]
+   [broadfcui.common.links :as links]
+   [broadfcui.common.icons :as icons]))
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/components/entity_details.cljs]: order/format

**Review by MattPutnam — APPROVED** [review]
👍, just cosmetics

File: src/cljs/main/broadfcui/page/workspace/analysis/igv.cljs [hunk 4]
```diff
@@ -38,3 +39,23 @@
    :refresh
```
Comment (zarsky-broad) [inline src/cljs/main/broadfcui/page/workspace/analysis/igv.cljs]: Note to self: probably actually need to expose this

File: src/cljs/main/broadfcui/page/workspace/analysis/igv.cljs [hunk 5]
```diff
@@ -38,3 +39,23 @@
...
    (fn [{:keys [props refs]}]
      (.createBrowser js/igv (@refs "container") (options (:tracks props))))})
+
+(react/defc IGVContainer
+  {:render
+   (fn [{:keys [props state]}]
+     (let [{:keys [deps-loaded? igv-loaded? error?]} @state]
+       [:div {}
+        [:link {:rel "stylesheet" :type "text/css" :href "https://igv.org/web/release/1.0.6/igv-1.0.6.css"}]
```
Comment (zarsky-broad) [inline src/cljs/main/broadfcui/page/workspace/analysis/igv.cljs]: Note to self: defonce to attach this to head

File: src/codemirror-deps.js [hunk 6]
```diff
@@ -0,0 +1,4 @@
+exports.CodeMirror = require('codemirror');
+require('codemirror/mode/clojure/clojure');
+require('codemirror/lib/codemirror.css');
+require('./styles/codemirror.scss');
```
Comment (zarsky-broad) [inline src/codemirror-deps.js]: Require base styles here

File: src/markdown-deps.js [hunk 7]
```diff
@@ -0,0 +1,3 @@
+exports.MarkdownIt = require('markdown-it');
+require('github-markdown-css/github-markdown.css');
+require('./styles/markdown.scss');
```
Comment (zarsky-broad) [inline src/markdown-deps.js]: Require base styles here

**Review by davidangb — APPROVED** [review]
:+1: working well for me!

**Pull request merged**